resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/zip/ZipFileStore.java

```diff
@@ -0,0 +1,559 @@
+		if (!(destination instanceof ZipFileStore)) {
+			super.move(destination, options, monitor);
+			return;
+		}
+		ZipFileStore destZipFileStore = (ZipFileStore) destination;
+
+		try (FileSystem srcFs = openZipFileSystem(); FileSystem destFs = destZipFileStore.openZipFileSystem()) {
+			Path srcPath = srcFs.getPath(this.path.toString());
+			Path destPath = destFs.getPath(destZipFileStore.path.toString());
+
+			if (destPath.getParent() != null) {
+				Files.createDirectories(destPath.getParent());
+			}
+
+			if (Files.isDirectory(srcPath)) {
+				moveDirectory(srcPath, destPath, srcFs, destFs);
+			} else {
+				Files.move(srcPath, destPath, StandardCopyOption.REPLACE_EXISTING);
+			}
+		} catch (IOException | URISyntaxException e) {
+			throw new CoreException(new Status(IStatus.ERROR, "org.eclipse.core.filesystem.zip", "Error moving entry within ZIP", e)); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+	}
+
+	private void moveDirectory(Path srcPath, Path destPath, FileSystem srcFs, FileSystem destFs) throws IOException {
+		// Ensure the destination directory structure is ready
+		if (destPath.getParent() != null) {
+			Files.createDirectories(destPath.getParent());
+		}
+
+		// Recursively move the contents
+		Files.walk(srcPath).forEach(source -> {
+			try {
+				Path destination = destPath.resolve(srcPath.relativize(source));
+				if (Files.isDirectory(source)) {
+					if (!Files.exists(destination)) {
+						Files.createDirectories(destination);
+					}
+				} else {
+					Files.move(source, destination, StandardCopyOption.REPLACE_EXISTING);
+				}
+			} catch (IOException e) {
+				throw new RuntimeException("Failed to move files", e); //$NON-NLS-1$
+			}
+		});
+
+		// Delete the source directory after moving its contents
+		Files.walk(srcPath).sorted(Comparator.reverseOrder()).forEach(pathToMove -> {
+			try {
+				Files.delete(pathToMove);
+			} catch (IOException e) {
+				throw new RuntimeException("Failed to delete original files after move", e); //$NON-NLS-1$
+			}
+		});
+	}
+
+	@Override
+	public InputStream openInputStream(int options, IProgressMonitor monitor) throws CoreException {
+		try {
+			ZipInputStream in = new ZipInputStream(rootStore.openInputStream(EFS.NONE, monitor));
+			ZipEntry entry = findEntry(path.toString(), in);
+			if (entry == null) {
+				throw new CoreException(Status.error("File not found: " + rootStore.toString())); //$NON-NLS-1$
+			}
+			if (entry.isDirectory()) {
+				throw new CoreException(Status.error("Resource is not a file: " + rootStore.toString())); //$NON-NLS-1$
+			}
+			return in;
+		} catch (IOException e) {
+			throw new CoreException(Status.error("Could not read file: " + rootStore.toString(), e)); //$NON-NLS-1$
+		}
+	}
+
+	@Override
+	public OutputStream openOutputStream(int options, IProgressMonitor monitor) {
+		// Creating a ByteArrayOutputStream to capture the data written to the
+		// OutputStream
+		ByteArrayOutputStream baos = new ByteArrayOutputStream() {
```

Comment:
```suggestion
		return new ByteArrayOutputStream() {
```

Reply:
yes this is cleaner. I added this change. Thank you!